srecord/input/filter/nibble_swap.h

```diff
@@ -0,0 +1,80 @@
+//
+// srecord filter that exchanges upper and lower nibble of each byte, e.g. "0xA6" => "0x6A"
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation; either version 3 of the License, or (at
+// your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+//
+
+#ifndef SRECORD_INPUT_FILTER_NIBBLE_SWAP_H
+#define SRECORD_INPUT_FILTER_NIBBLE_SWAP_H
+
+#include <srecord/input/filter.h>
+
+namespace srecord {
+
+/**
+  * The srecord::input_filter_nibble_swap class is used to represent the input state
+  * of a filter which reverses the order of bits in each data byte.
```

Comment:
Suggest 

`The srecord::input_filter_nibble_swap class is used to represent the input state
 of a filter which exchanges upper and lower nibble of each byte, e.g. "0xA6" => "0x6A"`

Reply:
Good catch, some copypasta leftovers there. will fix